### PR TITLE
Add scripts for converting Spoken Wikipedia to asr dataset

### PR DESCRIPTION
**Pull request by bene-ges**
Signed-off-by: Alexandra Antonova <aleksandraa@nvidia.com>

# What does this PR do ?

Scripts for converting Spoken Wikipedia to an asr dataset (download, preprocessing, ctc-segmentation, filtering).

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: scripts/dataset_processing/spoken_wikipedia/run.sh [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+rm -rf $OUTPUT_DIR
+rm -rf ${INPUT_DIR}_prepared
+mkdir ${INPUT_DIR}_prepared
+mkdir ${INPUT_DIR}_prepared/audio
+mkdir ${INPUT_DIR}_prepared/text
+python ${NEMO_PATH}/scripts/dataset_processing/spoken_wikipedia/preprocess.py --input_folder ${INPUT_DIR} --destination_folder ${INPUT_DIR}_prepared
+
+
+MODEL_FOR_SEGMENTATION="QuartzNet15x5Base-En"
```
Comment (ekmb) [inline scripts/dataset_processing/spoken_wikipedia/run.sh:49]: This is a duplication of https://github.com/NVIDIA/NeMo/blob/main/tools/ctc_segmentation/run_segmentation.sh and https://github.com/NVIDIA/NeMo/blob/main/tools/ctc_segmentation/run_filter.sh. How about calling these bash scripts with appropriate args from yours?

File: scripts/dataset_processing/spoken_wikipedia/run.sh [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## Download the Spoken Wikipedia corpus for English
+## Note, that there are some other other languages available
+wget https://corpora.uni-hamburg.de/hzsk/de/islandora/object/file:swc-2.0_en-with-audio/datastream/TAR/en-with-audio.tar .
+tar -xvf en-with-audio.tar
+
```
Comment (ekmb) [inline scripts/dataset_processing/spoken_wikipedia/run.sh:21]: based on https://nats.gitlab.io/swc/, this work should be cited:
```
@InProceedings{KHN16.518,
  author = {Arne K{\"o}hn and Florian Stegen and Timo Baumann},
  title = {Mining the Spoken Wikipedia for Speech Data and Beyond},
  booktitle = {Proceedings of the Tenth International Conference on Language Resources and Evaluation (LREC 2016)},
  year = {2016},
  month = {may},
  date = {23-28},
  location = {Portorož, Slovenia},
  editor = {Nicoletta Calzolari (Conference Chair) and Khalid Choukri and Thierry Declerck and Marko Grobelnik and Bente Maegaard and Joseph Mariani and Asuncion Moreno and Jan Odijk and Stelios Piperidis},
  publisher = {European Language Resources Association (ELRA)},
  address = {Paris, France},
  isbn = {978-2-9517408-9-1},
  islrn = {684-927-624-257-3/},
  language = {english}
 }
```

File: scripts/dataset_processing/spoken_wikipedia/run.sh [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+--CER_EDGE_THRESHOLD=${CER_EDGE_THRESHOLD} \
+--LEN_DIFF_RATIO_THRESHOLD=${LEN_DIFF_RATIO_THRESHOLD}
+
+python ${NEMO_PATH}/examples/asr/speech_to_text_eval.py \
+  dataset_manifest=${OUTPUT_DIR}/manifests/manifest_transcribed_metrics_filtered.json \
+  use_cer=True \
+  only_score_manifest=True
+
+python ${NEMO_PATH}/examples/asr/speech_to_text_eval.py \
+  dataset_manifest=${OUTPUT_DIR}/manifests/manifest_transcribed_metrics_filtered.json \
```
Comment (ekmb) [inline scripts/dataset_processing/spoken_wikipedia/run.sh:110]: i thought we had cer/wer after the filtering step, no?

**Review by ekmb — CHANGES_REQUESTED** [review]

File: scripts/dataset_processing/spoken_wikipedia/run.sh (hunk 3 above)
Comment (bene-ges) [reply scripts/dataset_processing/spoken_wikipedia/run.sh:110]: I do not see this information in the output.
I mean the WER/CER of the whole dataset, not each sentence.

File: scripts/dataset_processing/spoken_wikipedia/run.sh [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+##  │   ├── audio4.ogg
+##  │   ├── audiometa.txt
+##  │   ├── info.json
+##  │   ├── wiki.html
+##  │   ├── wiki.txt
+##  │   └── wiki.xml
+
+## Some rare folders are incomplete, these will be skipped during preprocessing.
+
+## path to NeMo repository, e.g. /home/user/nemo
```
[the remark this answers is not in the document]
Comment (bene-ges) [reply scripts/dataset_processing/spoken_wikipedia/run.sh:49]: done

File: scripts/dataset_processing/spoken_wikipedia/run.sh (hunk 2 above)
Comment (bene-ges) [reply scripts/dataset_processing/spoken_wikipedia/run.sh:21]: added

File: scripts/dataset_processing/spoken_wikipedia/preprocess.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    except Exception as e:
+        print("Error in sox: " + name)
+        print(e)
+        continue
+
+    # Then we need to clean the text
+    out_text = open(args.destination_folder + "/text/" + str(n) + ".txt", "w", encoding="utf-8")
+    with open(args.input_folder + "/" + name + "/wiki.txt", "r", encoding="utf-8") as f:
+        for line in f:
+            do_break = False
```
[the remark this answers is not in the document]
Comment (bene-ges) [reply scripts/dataset_processing/spoken_wikipedia/preprocess.py:119]: done

File: scripts/dataset_processing/spoken_wikipedia/preprocess.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    # Then we need to convert .ogg to .wav
+    output_wav_path = args.destination_folder + "/audio/" + str(n) + ".wav"
+    tfm = Transformer()
+    tfm.rate(samplerate=16000)
+    tfm.channels(n_channels=1)
+    if not os.path.exists(args.input_folder + "/" + name + "/wiki.txt"):
+        print("wiki.txt does not exist in " + name)
+        continue
+    try:
+        tfm.build(input_filepath=audio_path, output_filepath=output_wav_path)
```
[the remark this answers is not in the document]
Comment (bene-ges) [reply scripts/dataset_processing/spoken_wikipedia/preprocess.py:108]: removed this

File: scripts/dataset_processing/spoken_wikipedia/preprocess.py [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+        ##  │   ├── aligned.swc
+        ##  │   ├── audio1.ogg
+        ##  │   ├── audio2.ogg
+        ##  │   ├── audio3.ogg
+        ##  │   ├── audio4.ogg
+        ##  │   ├── audiometa.txt
+        ##  │   ├── info.json
+        ##  │   ├── wiki.html
+        ##  │   ├── wiki.txt
+        ##  │   └── wiki.xml
```
[the remark this answers is not in the document]
Comment (bene-ges) [reply scripts/dataset_processing/spoken_wikipedia/preprocess.py:76]: added this to the description

File: scripts/dataset_processing/spoken_wikipedia/run.sh (hunk 1 above)
Comment (bene-ges) [reply scripts/dataset_processing/spoken_wikipedia/run.sh:72]: done

**Review by ekmb — APPROVED** [review]

**Review by vsl9 — APPROVED** [review]

**Pull request merged**